### PR TITLE
CI: add some more jobs from macOS x86_64 for macOS arm64

### PR DESCRIPTION
**Pull request by pheiduck**
continue: #9565

### CODE REVIEW

**Comment by pheiduck** [comment]
the cmake gcc builds are failing because of an error which was fixed here: https://github.com/curl/curl/commit/935b1bd4544a23a91d68ffb9f86983e92747e9a5

**Comment by pheiduck** [comment]
@bagder @mback2k can be reviewed I skip the cmake gcc builds here because the failed to often.

**Comment by mback2k** [comment]
Please rework this into using a matrix build to avoid the huge amount of repetition.

**Comment by pheiduck** [comment]
> Please rework this into using a matrix build to avoid the huge amount of repetition.

I hope it still builds :crossed_fingers:

**Comment by pheiduck** [comment]
@mback2k The builds are failing what is wrong?

**Comment by mback2k** [comment]
This is YAML, so if you want to put a multi-line string somewhere you have to use a modifier after the colon `:`, e.g. `: |`, see:
https://yaml-multiline.info/

**Comment by pheiduck** [comment]
> This is YAML, so if you want to put a multi-line string somewhere you have to use a modifier after the colon `:`, e.g. `: |`, see: https://yaml-multiline.info/

hm, anyway still failing

**Comment by pheiduck** [comment]
@fkorotkov could you help out here?

File: .cirrus.yml [hunk 1]
```diff
@@ -133,49 +133,97 @@ windows_task:
...
 
 macos_task:
-  name: macOS arm64 normal
+  name: macOS arm64
   macos_instance:
-    image: ghcr.io/cirruslabs/macos-ventura-xcode:latest
-
+    image: ${container_img}
+    pkgs: ${prepare}
+    config: ${configure}
```
Comment (fkorotkov) [inline .cirrus.yml:140]: These two lines seems will be ignored.

File: .cirrus.yml [hunk 2]
```diff
@@ -133,49 +133,97 @@ windows_task:
...
+    - name: macOS arm64 torture-ftp
+      env:
+        container_img: ghcr.io/cirruslabs/macos-ventura-xcode:latest
+        prepare: echo libtool autoconf automake pkg-config nghttp2 openssl | xargs -Ix -n1 echo brew '"x"' > /tmp/Brewfile
+        configure: --enable-warnings --enable-werror --enable-debug --disable-shared --disable-threaded-resolver --with-openssl=$(brew --prefix openssl) --enable-websockets
+
+  prepare_script: |
+    ${pkgs} && brew update && for i in 1 2 3; do brew bundle install --no-lock --file /tmp/Brewfile && break || sleep 1; done && sudo -H python3 -m pip install --upgrade pip && sudo -H python3 -m pip install impacket
+  configure_script: |
+    autoreconf -fi && ./configure ${config}
```
Comment (fkorotkov) [inline .cirrus.yml:223]: I guess you want something like this:

```suggestion
  prepare_script: |
    ${prepare} && brew update && for i in 1 2 3; do brew bundle install --no-lock --file /tmp/Brewfile && break || sleep 1; done && sudo -H python3 -m pip install --upgrade pip && sudo -H python3 -m pip install impacket
  configure_script: |
    autoreconf -fi && ./configure ${configure}
```

**Comment by pheiduck** [comment]
@fkorotkov @mback2k the Problem is that the Brewfile is empty and on the build config variable is "the same problem"

**Comment by pheiduck** [comment]
`Error while parsing tasks: INVALID_ARGUMENT: yaml: line 154: mapping values are not allowed in this context!`

Not sure why this is showing

**Comment by dfandrich** [comment]
There's a no-break space between prepare: and | on line 152.

I find forcing CI YAML files through a YAML pretty-printer before submission useful to catch syntax issues. A one-liner like this works great for me:

    python3 -c 'import yaml,sys,pprint; print(pprint.PrettyPrinter(indent=4).pprint(yaml.load(sys.stdin,Loader=yaml.SafeLoader)))' < .cirrus.yml

(I posted this one-liner yesterday by e-mail, but it seems Github ate it).

**Comment by pheiduck** [comment]
> There's a no-break space between prepare: and | on line 152.
> 
> I find forcing CI YAML files through a YAML pretty-printer before submission useful to catch syntax issues. A one-liner like this works great for me:
> 
> ```
> python3 -c 'import yaml,sys,pprint; print(pprint.PrettyPrinter(indent=4).pprint(yaml.load(sys.stdin,Loader=yaml.SafeLoader)))' < .cirrus.yml
> ```
> 
> (I posted this one-liner yesterday by e-mail, but it seems Github ate it).

Doesn't solve it anyway, maybe we need to use it without matrixbuilds

**Comment by pheiduck** [comment]
I revert it back to the first commit have to revise it again later.

**Comment by mback2k** [comment]
Suggestion: I will make a matrix build out of the two jobs already there for macOS and then you can add the additional ones on top of that.

**Comment by mback2k** [comment]
I can take a look later, I guess you are escaping things wrong.

File: .cirrus.yml [hunk 3]
```diff
@@ -133,46 +133,81 @@ windows_task:
...
     MAKE_FLAGS: -j 4
-    CFLAGS: "-Wno-vla -mmacosx-version-min=10.9"
+    CFLAGS: "-Wno-vla -mmacosx-version-min=10.8"
 
   pkginstall_script:
-    - brew update
-    - echo libtool autoconf automake pkg-config nghttp2 | xargs -Ix -n1 echo brew '"x"' > /tmp/Brewfile
+    - echo libtool autoconf automake pkg-config ${install_packages} | xargs -Ix -n1 echo brew '"x"' > /tmp/Brewfile
     - brew update && for i in 1 2 3; do brew bundle install --no-lock --file /tmp/Brewfile && break || sleep 1; done
+    - export PKG_CONFIG_PATH=${prefix}
```
Comment (mback2k) [inline .cirrus.yml:205]: Try `$(eval echo $prefix)` instead of just `${prefix}`

Comment (pheiduck) [reply .cirrus.yml]: :/ doesn't work

**Comment by mback2k** [comment]
I guess the variable export does not survive till the next step.

**Comment by pheiduck** [comment]
> I guess the variable export does not survive till the next step.

Another solution is to use `brew link pkg --force` it will create symlinks in `/usr/local`

**Comment by mback2k** [comment]
Your latest commit has mixed/invalid indentation.

**Comment by mback2k** [comment]
I would recommend to add an install step like this:

for package in ${link_packages}; do brew link ${package}; done

**Comment by pheiduck** [comment]
> I would recommend to add an install step like this:
> 
> for package in linkpackages;dobrewlink{package} --force; done
> 
> This will do nothing if link_packages is empty or undefined.

My last 2 commit have solved the problem 🎉

**Comment by mback2k** [comment]
Generally LGTM, but please rebase onto master instead of merging from it and then squash your changes into one commit.

**Comment by pheiduck** [comment]
Wasn't that what I wound reopen https://github.com/curl/curl/pull/9676

**Comment by pheiduck** [comment]
Had to reopened it https://github.com/curl/curl/pull/9677 because it closed due a bad rebasing

**Comment by dfandrich** [comment]
I find forcing CI YAML files through a YAML pretty-printer before submission useful to catch syntax issues. A one-liner like this works great for me:

    python3 -c 'import yaml,sys,pprint; print(pprint.PrettyPrinter(indent=4).pprint(yaml.load(sys.stdin, Loader=yaml.SafeLoader)))' < appveyor.yml

**Pull request closed without merging**